### PR TITLE
Dynawo xml exporter

### PR DESCRIPTION
**Pull request by marcosmc**
write dynawo objects as XML files

### CODE REVIEW

File: dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoDynamicModels.java [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+    public static void writeDynamicModels(XMLStreamWriter writer, List<DynawoDynamicModel> dynamicModels)
+        throws XMLStreamException {
+        for (DynawoDynamicModel dynamicModel : dynamicModels) {
+            writeDynamicModel(writer, dynamicModel);
+        }
+    }
+
+    public static int countGeneratorConnections(List<DynawoDynamicModel> dynamicModels) {
+        return toIntExact(dynamicModels.stream().filter(dynamicModel -> Connection.class.isInstance(dynamicModel)
+            && ((Connection) dynamicModel).getVar2().equals("generator_omegaPu")).count());
```
Comment (rosiereflo) [inline dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoDynamicModels.java]: Small question (just to be sure I understand properly): how can we be sure that the generator name is "generator" and that the variable "omegaPu" exists for all generators? Is it forced somewhere ?

File: dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoJobs.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    }
+
+    private static void writeSimulation(XMLStreamWriter writer, DynawoSimulation simulation) throws XMLStreamException {
+        int startTime = simulation.getStartTime();
+        int stopTime = simulation.getStopTime();
+        boolean activeCriteria = simulation.isActiveCriteria();
+        writer.writeEmptyElement(DYN_URI, "simulation");
+        writer.writeAttribute("startTime", Integer.toString(startTime));
+        writer.writeAttribute("stopTime", Integer.toString(stopTime));
+        writer.writeAttribute("activateCriteria", Boolean.toString(activeCriteria));
```
Comment (rosiereflo) [inline dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoJobs.java]: We added very recently a "precision" attribute to specify the precision used for real numbers

File: dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoDynamicModels.java (hunk 1 above)
Comment (zamarrenolm) [reply dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoDynamicModels.java]: XML serialization should not try to "complete" the dynawo object model with additional default components and connections. Default models should be defined outside. XML Serialization should just write the required Dynawo input files from the set of Dynawo objects that has been built by the user. 

The user will be able to build Dynawo objects in different ways, using groovy for example, and eventually relying on a generic Dynamic Database (that will contain the dynamic information required and will have multiple implementations available).

File: dynawo/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+/target/
```
Comment (mathbagu) [inline dynawo/.gitignore]: I don't understand why we still have this file: we should have a .gitignore only in the root folder.

File: dynawo/src/main/groovy/com/powsybl/dynawo/dsl/job/DynawoJobDslLoader.groovy [hunk 4]
```diff
@@ -148,6 +149,12 @@ class DynawoJobDslLoader {
             this.activeCriteria = activeCriteria
             return this
         }
+
+        SimulationSpec precision(double precision) {
```
Comment (mathbagu) [inline dynawo/src/main/groovy/com/powsybl/dynawo/dsl/job/DynawoJobDslLoader.groovy]: Maybe you should also assert that precision is not NaN and positive?

File: dynawo/src/main/java/com/powsybl/dynawo/dyd/DynawoDynamicModel.java [hunk 5]
```diff
@@ -11,6 +11,6 @@
  */
 public interface DynawoDynamicModel {
 
-    String getId();
+    public abstract String getId();
```
Comment (mathbagu) [inline dynawo/src/main/java/com/powsybl/dynawo/dyd/DynawoDynamicModel.java]: As DynawoDynamicModel is an interface, revert this change

File: dynawo/src/main/java/com/powsybl/dynawo/par/DynawoParameterSet.java [hunk 6]
```diff
@@ -8,42 +8,50 @@
...
+        return Collections.unmodifiableMap(parameters);
     }
 
     public List<DynawoParameterTable> getParameterTables() {
         return Collections.unmodifiableList(parameterTables);
     }
 
-    public DynawoParameterSet addParameters(List<DynawoParameter> parameter) {
-        parameters.addAll(parameter);
+    public DynawoParameterSet addParameters(List<DynawoParameter> parameters) {
```
Comment (mathbagu) [inline dynawo/src/main/java/com/powsybl/dynawo/par/DynawoParameterSet.java]: To be consistent, either rename this method to `add`, or rename `add` to `addParameter`. I think I prefer the second option

File: dynawo/src/main/java/com/powsybl/dynawo/par/DynawoParameterSet.java [hunk 7]
```diff
@@ -8,42 +8,50 @@
...
         return this;
     }
 
     public DynawoParameterSet add(DynawoParameter parameter) {
         Objects.requireNonNull(parameter);
-        parameters.add(parameter);
+        if (parameters.containsKey(parameter.getName())) {
+            LOGGER.warn("parameter {} already exists, the last value entered is retained.", parameter.getName());
+        }
+        parameters.put(parameter.getName(), parameter);
```
Comment (mathbagu) [inline dynawo/src/main/java/com/powsybl/dynawo/par/DynawoParameterSet.java]: This is not efficient... if you look to the [put](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#put-K-V-) API, this method returns the old value if it exists, so you just have to store it in a variable and test if it's different from null:
```
if (parameters.put(parameter.getName(), parameter) != null) {
    LOGGER.warn("parameter {} already exists, the last value entered is retained.", parameter.getName());
}
```

File: dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoCurves.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * @author Marcos de Miguel <demiguelm at aia.es>
+ */
+public final class DynawoCurves {
+
+    private DynawoCurves() {
+    }
+
+    public static void writeCurves(XMLStreamWriter writer, List<DynawoCurve> curves) throws XMLStreamException {
```
Comment (mathbagu) [inline dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoCurves.java]: Assert that writer and curves are not null.

File: dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoCurves.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        String model = curve.getModel();
+        String variable = curve.getVariable();
+        writeCurve(writer, model, variable);
+    }
+
+    private static void writeCurve(XMLStreamWriter writer, String model, String variable) throws XMLStreamException {
+        writer.writeEmptyElement("curve");
+        writer.writeAttribute("model", model);
+        writer.writeAttribute("variable", variable);
+    }
```
Comment (mathbagu) [inline dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoCurves.java]: Is it really necessary to split this method in 3 part? Moreover the 2 other parts are private so unusable from the outside

File: dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoDynamicModels.java [hunk 10]
```diff
@@ -0,0 +1,314 @@
...
+        } else if (MacroConnector.class.isInstance(dynamicModel)) {
+            writeMacroConnector(writer, (MacroConnector) dynamicModel);
+        } else if (MacroConnection.class.isInstance(dynamicModel)) {
+            writeMacroConnection(writer, (MacroConnection) dynamicModel);
+        } else if (MacroStaticReference.class.isInstance(dynamicModel)) {
+            writeMacroStaticReference(writer, (MacroStaticReference) dynamicModel);
+        } else {
+            throw new PowsyblException("DynamicModel class " + dynamicModel.getClass().getName() + " is not allowed");
+        }
+    }
```
Comment (mathbagu) [inline dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoDynamicModels.java]: To avoid this kind of design and complexity, we should create a map to associate DynamicModel and export function. What do you think?

File: dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoDynamicModels.java [hunk 11]
```diff
@@ -0,0 +1,314 @@
...
+    private static void writeBlackBoxModel(XMLStreamWriter writer, BlackBoxModel dynamicModel)
+        throws XMLStreamException {
+        String id = dynamicModel.getId();
+        String lib = dynamicModel.getLib();
+        String file = dynamicModel.getParametersFile();
+        String paramId = dynamicModel.getParametersId();
+        String staticId = dynamicModel.getStaticId();
+        List<StaticRef> staticRefs = dynamicModel.getStaticRefs();
+        List<DydComponent> macroStaticRefs = dynamicModel.getMacroStaticRefs();
+        writeBlackBoxModel(writer, id, lib, file, paramId, staticId, staticRefs, macroStaticRefs);
```
Comment (mathbagu) [inline dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoDynamicModels.java]: why do you extract information from a blackBoxModel to write it to XML (through another method), instead of writing it directly?

File: dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoDynamicModels.java [hunk 12]
```diff
@@ -0,0 +1,314 @@
...
+        throws XMLStreamException {
+        String id = dynamicModel.getId();
+        String templateId = dynamicModel.getTemplateId();
+        String file = dynamicModel.getParametersFile();
+        String paramId = dynamicModel.getParametersId();
+        writer.writeEmptyElement(DYN_URI, "modelTemplateExpansion");
+        writer.writeAttribute("id", id);
+        writer.writeAttribute("templateId", templateId);
+        writer.writeAttribute("parFile", file);
+        writer.writeAttribute("parId", paramId);
```
Comment (mathbagu) [inline dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoDynamicModels.java]: Do you have to close the current element here?

File: dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoDynamicModels.java [hunk 13]
```diff
@@ -0,0 +1,314 @@
...
+    private static String translate(String token, Injection<?> injection) {
+        return token.replace("#ID#", injection.getId()).replace("#IDBUS#",
+            injection.getTerminal().getBusBreakerView() != null ? injection.getTerminal().getBusBreakerView().getBus().getId()
+                : injection.getTerminal().getBusView().getBus().getId());
+    }
+
+    private static String translate(String token, int id) {
+        return token.replace("#NUM#", Integer.toString(id));
+    }
+}
```
Comment (mathbagu) [inline dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoDynamicModels.java]: A general remark: is there necessary to split each write methods in part? Is these new methods use elsewhere?

File: dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoInputs.java [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+                crvXmlWriter.writeEndDocument();
+            } finally {
+                jobXmlWriter.close();
+                dydXmlWriter.close();
+                parXmlWriter.close();
+                parSolverXmlWriter.close();
+                crvXmlWriter.close();
+            }
+        }
+    }
```
Comment (mathbagu) [inline dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoInputs.java]: Is is better to create 5 methods, one per writer? I think it's not really readable to do like this and avoid mistakes

File: dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoInputs.java [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+    public static String getCopyrightText() {
+        return String.join(System.lineSeparator(),
+            "    Copyright (c) 2015-2019, RTE (http://www.rte-france.com)",
+            "    See AUTHORS.txt",
+            "    All rights reserved.",
+            "    This Source Code Form is subject to the terms of the Mozilla Public",
+            "    License, v. 2.0. If a copy of the MPL was not distributed with this",
+            "    file, you can obtain one at http://mozilla.org/MPL/2.0/.",
+            "    SPDX-License-Identifier: MPL-2.0");
+    }
```
Comment (mathbagu) [inline dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoInputs.java]: Once again: I think the files serialized by our code should not be copyrighted RTE or under the MPL license. What do you think? For instance, all IIDM files we serialize are not our property.

File: dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoJobs.java [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+    private static void writeAppender(XMLStreamWriter writer, LogAppender appender) throws XMLStreamException {
+        String tag = appender.getTag();
+        String file = appender.getFile();
+        String lvlFilter = appender.getLvlFilter();
+        writer.writeEmptyElement(DYN_URI, "appender");
+        writer.writeAttribute("tag", tag);
+        writer.writeAttribute("file", file);
+        writer.writeAttribute("lvlFilter", lvlFilter);
+    }
+}
```
Comment (mathbagu) [inline dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoJobs.java]: Is it a good design to extract data instead of using getter directly? It makes the code unnecessarily longer

File: dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoSimulationParameters.java [hunk 17]
```diff
@@ -0,0 +1,112 @@
...
+        writer.writeEmptyElement("reference");
+        writer.writeAttribute("name", name);
+        writer.writeAttribute("origData", origData);
+        writer.writeAttribute("origName", origName);
+        writer.writeAttribute("type", type);
+        if (componentId != null) {
+            writer.writeAttribute("componentId", componentId);
+        }
+    }
+}
```
Comment (mathbagu) [inline dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoSimulationParameters.java]: Same remark here

File: dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoSolverParameters.java [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+        writeParameter(writer, type, name, value);
+    }
+
+    private static void writeParameter(XMLStreamWriter writer, String type, String name, String value)
+        throws XMLStreamException {
+        writer.writeEmptyElement("par");
+        writer.writeAttribute("type", type);
+        writer.writeAttribute("name", name);
+        writer.writeAttribute("value", value);
+    }
```
Comment (mathbagu) [inline dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoSolverParameters.java]: Same remark about splitting methods and extracting values from objects

File: dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoXmlExporter.java [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+public class DynawoXmlExporter {
+
+    private static final String DEFAULT_DYNAWO_CASE_NAME = "nrt/data/IEEE14/IEEE14_BasicTestCases/IEEE14_DisconnectLine/IEEE14.jobs";
+
+    public DynawoXmlExporter() {
+        this(PlatformConfig.defaultConfig());
+    }
+
+    public DynawoXmlExporter(PlatformConfig platformConfig) {
+        this.platformConfig = platformConfig;
```
Comment (mathbagu) [inline dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoXmlExporter.java]: Assert that platformConfig is not null

File: dynawo/src/test/java/com/powsybl/dynawo/xml/DynawoExporterTest.java [hunk 20]
```diff
@@ -0,0 +1,343 @@
...
+
+    @Test
+    public void testCurve() throws IOException, XMLStreamException {
+        writeToDslFile("curve {",
+            "    model 'model'",
+            "    variable 'variable'",
+            "}");
+
+        exporter.export(network, new GroovyDslDynawoInputProvider(dslFile), tmpDir);
+        InputStream is = Files.newInputStream(tmpDir.resolve("dynawoModel.crv"));
```
Comment (mathbagu) [inline dynawo/src/test/java/com/powsybl/dynawo/xml/DynawoExporterTest.java]: you should close this inputStream even it's a in-memory one, using try-with-resources pattern

File: dynawo/src/test/java/com/powsybl/dynawo/xml/DynawoExporterTest.java [hunk 21]
```diff
@@ -0,0 +1,343 @@
...
+            "            var 'var'",
+            "            staticVar 'staticVar'",
+            "        }",
+            "    }",
+            "}",
+            "macroConnection {",
+            "    connector 'connector'",
+            "    id1 'id1'",
+            "    id2 'id2'",
+            "}");
```
Comment (mathbagu) [inline dynawo/src/test/java/com/powsybl/dynawo/xml/DynawoExporterTest.java]: You should extract code like this in a resource file.

File: dynawo/src/test/java/com/powsybl/dynawo/xml/DynawoExporterTest.java [hunk 22]
```diff
@@ -0,0 +1,343 @@
...
+
+        exporter.export(network, new GroovyDslDynawoInputProvider(dslFile), tmpDir);
+        InputStream is = Files.newInputStream(tmpDir.resolve("dynawoModel.dyd"));
```
Comment (mathbagu) [inline dynawo/src/test/java/com/powsybl/dynawo/xml/DynawoExporterTest.java]: Same remark here

File: dynawo/src/test/java/com/powsybl/dynawo/xml/DynawoExporterTest.java [hunk 23]
```diff
@@ -0,0 +1,343 @@
...
+        ReadOnlyDataSource ds = gm.dataSource();
+        Network n = i.importData(ds, NetworkFactory.findDefault(), params);
+        return n;
+    }
+
+    private Network network;
+    private DynawoInputProvider dynawoProvider;
+    private Path dslFile;
+    private DynawoXmlExporter exporter;
+}
```
Comment (mathbagu) [inline dynawo/src/test/java/com/powsybl/dynawo/xml/DynawoExporterTest.java]: Global remark: extract DSL content to resource files, and close the inputStream after usage

File: dynawo/src/test/resources/dynawoModel.crv [hunk 24]
```diff
@@ -0,0 +1,15 @@
...
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, you can obtain one at http://mozilla.org/MPL/2.0/.
+    SPDX-License-Identifier: MPL-2.0
+
+    This file is part of Dynawo, an hybrid C++/Modelica open source time domain
+    simulation tool for power systems.-->
+<curvesInput xmlns="http://www.rte-france.com/dynawo">
+    <!--Curves for scenario-->
+    <curve model="model" variable="variable"/>
+</curvesInput>
```
Comment (mathbagu) [inline dynawo/src/test/resources/dynawoModel.crv]: Add an empty line

File: dynawo/src/test/resources/dynawoModel.dyd [hunk 25]
```diff
@@ -0,0 +1,41 @@
...
+    <dyn:connect id1="id1" var1="var1" id2="id2" var2="var2"/>
+    <dyn:initConnect id1="id1" var1="var1" id2="id2" var2="var2"/>
+    <dyn:macroConnector id="mcid">
+        <dyn:connect var1="var1" var2="var2"/>
+    </dyn:macroConnector>
+    <dyn:macroStaticReference id="msrid">
+        <dyn:staticRef var="var" staticVar="staticVar"/>
+    </dyn:macroStaticReference>
+    <dyn:macroConnect connector="connector" id1="id1" id2="id2"/>
+</dyn:dynamicModelsArchitecture>
```
Comment (mathbagu) [inline dynawo/src/test/resources/dynawoModel.dyd]: Same remark

File: dynawo/src/test/resources/dynawoModel.jobs [hunk 26]
```diff
@@ -0,0 +1,32 @@
...
+        <dyn:outputs directory="directory">
+            <dyn:dumpInitValues local="true" global="true"/>
+            <dyn:curves inputFile="curve" exportMode="CSV"/>
+            <dyn:timeline exportMode="TXT"/>
+            <dyn:logs>
+                <dyn:appender tag="tag" file="file" lvlFilter="lvlFilter"/>
+            </dyn:logs>
+        </dyn:outputs>
+    </dyn:job>
+</dyn:jobs>
```
Comment (mathbagu) [inline dynawo/src/test/resources/dynawoModel.jobs]: Same remark

File: dynawo/src/test/resources/dynawoModel.par [hunk 27]
```diff
@@ -0,0 +1,20 @@
...
+    simulation tool for power systems.-->
+<parametersSet xmlns="http://www.rte-france.com/dynawo">
+    <set id="1">
+        <par type="type2" name="name2" value="value2"/>
+        <reference name="name1" origData="origData1" origName="origName1" type="type1" componentId="componentId1"/>
+        <parTable type="nameTable" name="typeTable">
+            <row row="1" column="1" value="valueRow"/>
+        </parTable>
+    </set>
+</parametersSet>
```
Comment (mathbagu) [inline dynawo/src/test/resources/dynawoModel.par]: Same remark

File: dynawo/src/test/resources/nordic32/dynawoModel.crv [hunk 28]
```diff
@@ -0,0 +1,20 @@
...
+    <curve model="NETWORK" variable="_N1011____TN_Upu_value"/>
+    <curve model="_G10______SM" variable="generator_omegaPu"/>
+    <curve model="_G10______SM" variable="generator_PGen"/>
+    <curve model="_G10______SM" variable="generator_QGen"/>
+    <curve model="_G10______SM" variable="generator_UStatorPu"/>
+    <curve model="_G10______SM" variable="voltageRegulator_UcEfdPu"/>
+    <curve model="_G10______SM" variable="voltageRegulator_EfdPu"/>
+    <curve model="_N1011____EC" variable="load_PPu"/>
+    <curve model="_N1011____EC" variable="load_QPu"/>
+</curvesInput>
```
Comment (mathbagu) [inline dynawo/src/test/resources/nordic32/dynawoModel.crv]: Same remakr

File: dynawo/src/test/resources/nordic32/dynawoModel.dyd [hunk 29]
```diff
@@ -0,0 +1,22 @@
...
+    <dyn:blackBoxModel id="DISCONNECT_LINE" lib="EventQuadripoleDisconnection" parFile="dynawoModel.par" parId="5"/>
+    <dyn:connect id1="_N1011____EC" var1="load_terminal" id2="NETWORK" var2="_N1011____TN_ACPIN"/>
+    <dyn:connect id1="OMEGA_REF" var1="omega_grp_0" id2="_G10______SM" var2="generator_omegaPu"/>
+    <dyn:connect id1="OMEGA_REF" var1="omegaRef_grp_0" id2="_G10______SM" var2="generator_omegaRefPu"/>
+    <dyn:connect id1="OMEGA_REF" var1="numcc_node_0" id2="NETWORK" var2="@_G10______SM@@NODE@_numcc"/>
+    <dyn:connect id1="OMEGA_REF" var1="running_grp_0" id2="_G10______SM" var2="generator_running"/>
+    <dyn:connect id1="_G10______SM" var1="generator_terminal" id2="NETWORK" var2="@_G10______SM@@NODE@_ACPIN"/>
+    <dyn:connect id1="_G10______SM" var1="generator_switchOffSignal1" id2="NETWORK" var2="@_G10______SM@@NODE@_switchOff"/>
+    <dyn:connect id1="DISCONNECT_LINE" var1="event_state1_value" id2="NETWORK" var2="_N1011___-N1013___-1_AC_state_value"/>
+</dyn:dynamicModelsArchitecture>
```
Comment (mathbagu) [inline dynawo/src/test/resources/nordic32/dynawoModel.dyd]: same remark

File: dynawo/src/test/resources/nordic32/dynawoModel.jobs [hunk 30]
```diff
@@ -1,35 +1,30 @@
...
+            <dyn:curves inputFile="dynawoModel.crv" exportMode="CSV"/>
+            <dyn:timeline exportMode="TXT"/>
+            <dyn:logs>
+                <dyn:appender tag="" file="dynawo.log" lvlFilter="DEBUG"/>
+                <dyn:appender tag="COMPILE" file="dynawoCompiler.log" lvlFilter="DEBUG"/>
+                <dyn:appender tag="MODELER" file="dynawoModeler.log" lvlFilter="DEBUG"/>
+            </dyn:logs>
+        </dyn:outputs>
+    </dyn:job>
+</dyn:jobs>
```
Comment (mathbagu) [inline dynawo/src/test/resources/nordic32/dynawoModel.jobs]: same remark

File: dynawo/src/test/resources/nordic32/dynawoModel.par [hunk 31]
```diff
@@ -0,0 +1,112 @@
...
+        <par type="DOUBLE" name="generator_XpqPu" value="0.39300000000000002"/>
+        <par type="DOUBLE" name="generator_DPu" value="0"/>
+        <par type="DOUBLE" name="generator_XTfPu" value="0.1"/>
+    </set>
+    <set id="5">
+        <par type="BOOL" name="event_disconnectOrigin" value="false"/>
+        <par type="DOUBLE" name="event_tEvent" value="1"/>
+        <par type="BOOL" name="event_disconnectExtremity" value="true"/>
+    </set>
+</parametersSet>
```
Comment (mathbagu) [inline dynawo/src/test/resources/nordic32/dynawoModel.par]: same remark

File: dynawo/src/test/resources/nordic32/dynawoModel.xiidm [hunk 32]
```diff
@@ -0,0 +1,685 @@
...
+    <iidm:line id="_N4045___-N4062___-1_AC" name="N4045   -N4062   -1" r="17.6" x="128.0" g1="0.0" b1="7.5E-4" g2="0.0" b2="7.5E-4" bus1="_N4045____TN" connectableBus1="_N4045____TN" voltageLevelId1="_N4045____VL" bus2="_N4062____TN" connectableBus2="_N4062____TN" voltageLevelId2="_N4062____VL">
+        <iidm:currentLimits1 permanentLimit="1443.38"/>
+    </iidm:line>
+    <iidm:line id="_N4046___-N4047___-1_AC" name="N4046   -N4047   -1" r="1.6" x="24.0" g1="0.0" b1="1.5625E-4" g2="0.0" b2="1.5625E-4" bus1="_N4046____TN" connectableBus1="_N4046____TN" voltageLevelId1="_N4046____VL" bus2="_N4047____TN" connectableBus2="_N4047____TN" voltageLevelId2="_N4047____VL">
+        <iidm:currentLimits1 permanentLimit="1443.38"/>
+    </iidm:line>
+    <iidm:line id="_N4061___-N4062___-1_AC" name="N4061   -N4062   -1" r="3.2" x="32.0" g1="0.0" b1="1.875E-4" g2="0.0" b2="1.875E-4" bus1="_N4061____TN" connectableBus1="_N4061____TN" voltageLevelId1="_N4061____VL" bus2="_N4062____TN" connectableBus2="_N4062____TN" voltageLevelId2="_N4062____VL">
+        <iidm:currentLimits1 permanentLimit="1443.38"/>
+    </iidm:line>
+    <iidm:line id="_N4062___-N4063_
```
Comment (mathbagu) [inline dynawo/src/test/resources/nordic32/dynawoModel.xiidm]: same remark

File: dynawo/src/test/resources/nordic32/nordic32.groovy [hunk 33]
```diff
@@ -0,0 +1,676 @@
...
+    variable 'voltageRegulator_EfdPu'
+}
+curve {
+    model '_N1011____EC'
+    variable 'load_PPu'
+}
+curve {
+    model '_N1011____EC'
+    variable 'load_QPu'
+}
```
Comment (mathbagu) [inline dynawo/src/test/resources/nordic32/nordic32.groovy]: same remark

File: dynawo/src/test/resources/nordic32/solvers.par [hunk 34]
```diff
@@ -1,66 +1,18 @@
...
-  </set>
+    <set id="2">
+        <par type="DOUBLE" name="initStep" value="0.000001"/>
+        <par type="DOUBLE" name="absAccuracy" value="1e-4"/>
+        <par type="DOUBLE" name="minStep" value="0.000001"/>
+        <par type="DOUBLE" name="relAccuracy" value="1e-4"/>
+        <par type="INT" name="order" value="2"/>
+        <par type="DOUBLE" name="maxStep" value="10"/>
+    </set>
 </parametersSet>
```
Comment (mathbagu) [inline dynawo/src/test/resources/nordic32/solvers.par]: same remark

File: dynawo/src/test/resources/solvers.par [hunk 35]
```diff
@@ -0,0 +1,16 @@
...
+    file, you can obtain one at http://mozilla.org/MPL/2.0/.
+    SPDX-License-Identifier: MPL-2.0
+
+    This file is part of Dynawo, an hybrid C++/Modelica open source time domain
+    simulation tool for power systems.-->
+<parametersSet xmlns="http://www.rte-france.com/dynawo">
+    <set id="1">
+        <par type="type" name="name" value="value"/>
+    </set>
+</parametersSet>
```
Comment (mathbagu) [inline dynawo/src/test/resources/solvers.par]: Same remark

File: pom.xml [hunk 36]
```diff
@@ -470,6 +471,11 @@
                 <artifactId>powsybl-commons</artifactId>
                 <version>${powsyblcore.version}</version>
             </dependency>
+            <dependency>
+                <groupId>com.powsybl</groupId>
+                <artifactId>powsybl-config-classic</artifactId>
+                <version>${powsyblcore.version}</version>
+            </dependency>
```
Comment (mathbagu) [inline pom.xml]: Remove this dependency: you should not use this?

File: dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoInputs.java [hunk 37]
```diff
@@ -0,0 +1,175 @@
...
+        prepareCrvFile(network, inputProvider, workingDir);
+    }
+
+    public static void prepareJobFile(Network network, DynawoInputProvider inputProvider, Path workingDir)
+        throws IOException, XMLStreamException {
+        Path jobFile = workingDir.resolve(JOBS_FILENAME);
+        XMLOutputFactory output = XMLOutputFactory.newInstance();
+        try (Writer jobWriter = Files.newBufferedWriter(jobFile, StandardCharsets.UTF_8)) {
+            XMLStreamWriter jobXmlWriter = output.createXMLStreamWriter(jobWriter);
+            try {
```
Comment (mathbagu) [inline dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoInputs.java]: Small problem with try-with-resources: the idea is to not forget to close the writer. So the jobWriter, and the XmlWriter should be declared in the try.
```
try (XmlStreamWriter jobXmlWriter = output.createXmlStreamWriter(Files.newBufferedWriter(jobFile, StandardCharsets.UTF-8))) {
   ....
}
```

File: dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoInputs.java (hunk 15 above)
Comment (zamarrenolm) [reply dynawo/src/main/java/com/powsybl/dynawo/xml/DynawoInputs.java]: Agree. I assume we were trying to reproduce the content to the reference test cases.
